modules/web-console/frontend/app/modules/agent/AgentManager.service.ts

```diff
@@ -104,15 +104,27 @@ class ConnectionState {
 
         if (_.isEmpty(this.clusters))
             this.cluster = null;
-        else if (_.isNil(this.cluster))
-            this.cluster = _.head(clusters);
+        else if (_.isNil(this.cluster)) {
+            const restoredCluster = _.find(clusters, {id: localStorage.clusterId});
+
+            if (restoredCluster)
+                this.cluster = restoredCluster;
```

Comment:
this.cluster = restoredCluster || _.head(clusters);

Reply:
Simplified.